File: src/main/java/com/gitlab/daring/image/MainContext.java
package com.gitlab.daring.image;

import com.gitlab.daring.image.event.VoidEvent;

public class MainContext implements AutoCloseable {

    static final MainContext Instance = new MainContext();

    public static MainContext mainContext() {
        return Instance;
    }

    public final VoidEvent closeEvent = new VoidEvent();

    public void close() {
        closeEvent.fire();
    }

    private MainContext() {
    }

}

File: src/main/kotlin/com/gitlab/daring/image/MainContext.kt
package com.gitlab.daring.image

import com.gitlab.daring.image.command.CommandRegistry
import com.gitlab.daring.image.event.UnitEvent
import java.util.*

object MainContext : AutoCloseable {

    @JvmField
    val closeEvent = UnitEvent()
    @JvmField
    val timer = Timer(true)
    @JvmField
    val commandRegistry = CommandRegistry()

    override fun close() {
        timer.cancel()
        closeEvent.fire()
    }

}